ping.go

```diff
@@ -473,7 +473,17 @@ func (p *Pinger) recvICMP(
 			return nil
 		default:
 			// ICMP messages have an 8-byte header.
-			bytes := make([]byte, p.Size+8)
+			length := p.Size + 8
+			// But on Windows we have to deserialise the whole
+			// IP packet, not just the message in its payload.
+			if runtime.GOOS == "windows" {
```

Comment:
I think this might be a better case to use build flags. Refactor this length handler into a private method, and have one version for windows, one for not.

Reply:
Yeah that sounds better. I'll get on that